crates/aranya-crypto/src/aranya.rs

```diff
@@ -465,3 +465,87 @@ where
         d.deserialize_bytes(EncapVisitor(PhantomData))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use spideroak_crypto::rust;
+
+    use super::*;
+    use crate::{default::DhKemP256HkdfSha256, test_util::TestCs};
+
+    #[allow(dead_code)]
+    type CS = TestCs<
+        rust::Aes256Gcm,
+        rust::Sha256,
+        rust::HkdfSha512,
+        DhKemP256HkdfSha256,
+        rust::HmacSha512,
+        spideroak_crypto::ed25519::Ed25519,
+    >;
+
+    /// Golden test for [`IdentityKey::id`] (DeviceId) with deterministic keys.
+    #[test]
+    fn test_device_id() {
+        use spideroak_crypto::import::Import;
+        
+        let tests = [
+            (
+                [0u8; 32],
+                "4WvDDSvXv8Cqj85xHAkesJo3uVLy3LRczojr98KBNG8e",
+            ),
+            (
+                [1u8; 32],
+                "7czZtx4VW5V6K3eu59rSBJKiqeVZjDX2gyJ4tkJqvqEM",
+            ),
+            (
+                [0xFF; 32],
+                "DqAAPgZfS8KCo1W4M7C9A36SRXwoMKeLox5qzS1L7Bki",
+            ),
+        ];
+        
+        for (i, (key_bytes, expected)) in tests.iter().enumerate() {
+            let sk = IdentityKey::<CS>::from_inner(Import::import(&key_bytes[..]).unwrap());
+            let got = sk.id().unwrap();
+            let want = DeviceId::decode(expected).unwrap();
+            assert_eq!(got, want, "test case #{i}");
+        }
+    }
+
+    /// Golden test for [`SigningKey::id`] (SigningKeyId) with deterministic keys.
+    #[test]
+    fn test_signing_key_id() {
+        use spideroak_crypto::import::Import;
+        
+        let tests = [
+            (
+                [0u8; 32],
+                "DTSYenur9akPjgwULDLDhPdaZ9pmEKq7pYJ6vvST2kWs",
+            ),
+            (
+                [1u8; 32],
+                "6fuCv2oLzQzK2sBDyKh5eHcTZ2b6uJXC5txoNwqyWuQ5",
+            ),
+            (
+                [0xFF; 32],
+                "EgLiA9CxqVLKgdgLen9N5JJzh96vxDDrPXTGdBnpFzYe",
+            ),
+        ];
+        
+        for (i, (key_bytes, expected)) in tests.iter().enumerate() {
+            let sk = SigningKey::<CS>::from_inner(Import::import(&key_bytes[..]).unwrap());
+            let got = sk.id().unwrap();
+            let want = SigningKeyId::decode(expected).unwrap();
+            assert_eq!(got, want, "test case #{i}");
+        }
+    }
+
+    /// Golden test for [`EncryptionKey::id`] (EncryptionKeyId) with deterministic keys.
+    /// TODO: Implement deterministic key generation for KEM keys
+    #[test]
+    #[ignore]
+    fn test_encryption_key_id() {
+        // This test is temporarily disabled because KEM key deterministic generation
+        // requires a different approach than Ed25519 signing keys.
+        // TODO: Research how to create deterministic P256 private keys from byte arrays
```

Comment:
it's in ASN.1 uncompressed format: `0x4 || x || y` where `x` and `y` are the 32-byte coordinates